manager/src/test/java/com/redhat/service/smartevents/manager/api/user/validators/processors/CloudProviderConstraintValidatorTest.java

```diff
@@ -0,0 +1,176 @@
+                "The requested Region '" + br.getRegion() + "' is not valid.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    @Test
+    public void validate_emptyProviderAndRegion() {
+
+        BridgeRequest br = new BridgeRequest(DEFAULT_BRIDGE_NAME, "", "");
+
+        Set<String> violations = Set.of(
+                "Cloud Region cannot be null or empty.",
+                "Cloud Provider cannot be null or empty.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    @Test
+    public void validate_nullProviderAndNullRegion() {
+        BridgeRequest br = new BridgeRequest(DEFAULT_BRIDGE_NAME, null, null);
+
+        Set<String> violations = Set.of(
+                "Cloud Region cannot be null or empty.",
+                "Cloud Provider cannot be null or empty.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    @Test
+    public void validate_invalidCloudRegion() {
+        String invalidRegion = "dodgyRegion";
+        BridgeRequest br = new BridgeRequest(DEFAULT_BRIDGE_NAME, DEFAULT_CLOUD_PROVIDER, invalidRegion);
+
+        Set<String> violations = Set.of(
+                "The requested Region '" + invalidRegion + "' is not valid.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    @Test
+    public void validate_cloudProviderNotEnabled() {
+
+        String disabledProviderId = "gcp";
+        CloudProvider cp = cloudProviderDAO.findById(disabledProviderId);
+        assertThat(cp.isEnabled()).isFalse();
+
+        CloudRegion region = cp.getRegions().get(0);
+        BridgeRequest br = new BridgeRequest(DEFAULT_BRIDGE_NAME, cp.getId(), region.getName());
+
+        Set<String> violations = Set.of(
+                "The requested Cloud Provider '" + disabledProviderId + "' is not enabled.",
+                "The requested Region '" + region.getName() + "' is not enabled.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    @Test
+    public void validate_regionNotEnabled() {
+
+        String disabledRegionName = "eu-west-1";
+        CloudProvider cp = cloudProviderDAO.findById(DEFAULT_CLOUD_PROVIDER);
+        CloudRegion region = cp.getRegionByName(disabledRegionName).get();
+        assertThat(region.isEnabled()).isFalse();
+
+        BridgeRequest br = new BridgeRequest(DEFAULT_BRIDGE_NAME, DEFAULT_CLOUD_PROVIDER, region.getName());
+        Set<String> violations = Set.of(
+                "The requested Region '" + disabledRegionName + "' is not enabled.");
+
+        validateConstraintMessage(br, violations);
+    }
+
+    private void validateConstraintMessage(BridgeRequest br, Set<String> expectedMessages) {
+        Set<ConstraintViolation<BridgeRequest>> violations = validatorFactory.getValidator().validate(br);
+        assertThat(violations).hasSize(expectedMessages.size());
+
+        violations.forEach((v) -> assertThat(expectedMessages).contains(v.getMessage()));
```

Comment:
Optional, this can be simplified (due to fluent api single `assertThat` is enough)

```java
assertThat(violations)
                .hasSize(expectedMessages.size())
                .allSatisfy(violation -> expectedMessages.contains(violation.getMessage()));
```

But feel free to keep it as it is if you wish.

Reply:
Thanks @jomarko  - That's a nice tip!